### PR TITLE
Add documentation to all parameters shown by --help

### LINKED ISSUE #3774: Add documentation to all parameters shown by `--help`
Running nethermind with `--help` shows these command line parameters with missing documentation:

```
  --Discovery.Bootnodes                                        <missing documentation>
  --NoCategory.BaseDbPath                                      <missing documentation>
  --NoCategory.CliSwitchLocal                                  <missing documentation>
  --NoCategory.ConfigsDirectory                                <missing documentation>
  --NoCategory.CorsOrigins                                     <missing documentation>
  --NoCategory.DataDir                                         <missing documentation>
  --NoCategory.EnodeIpAddress                                  <missing documentation>
  --NoCategory.HiveEnabled                                     <missing documentation>
  --NoCategory.LoggerConfigSource                              <missing documentation>
  --NoCategory.LogLevel                                        <missing documentation>
  --NoCategory.MonitoringGroup                                 <missing documentation>
  --NoCategory.MonitoringJob                                   <missing documentation>
  --NoCategory.PluginsDirectory                                <missing documentation>
  --NoCategory.Url                                             <missing documentation>
  ```

Please add that documentation.

### PR DESCRIPTION
**Pull request by dceleda**
Resolves #3774

- Add description attribute to all parameters from the _NoCategory_ category.
- Don't add NoCategory configs dynamically as Command Line options, because some are already added explicitly in _Programs.cs_ and some are only Environment Variable configs
- Add additional information about available Environment Variables when `--help` is run
```
...
  --Wallet.DevAccounts                                         Number of auto-generted dev accounts to work with. Dev accounts will have private keys from 00...01 to 00..n (DEFAULT: 10)

Configurable Environment Variables:
NETHERMIND_CLI_SWITCH_LOCAL - Defines host value for CLI function "switchLocal". (DEFAULT: http://localhost)
NETHERMIND_CORS_ORIGINS - Defines CORS origins for JSON RPC. (DEFAULT: *)
NETHERMIND_ENODE_IPADDRESS - Sets the external IP for the node. (DEFAULT: )
NETHERMIND_HIVE_ENABLED - Enables Hive plugin used for executing Hive Ethereum Tests. (DEFAULT: false)
NETHERMIND_MONITORING_GROUP - Sets the default group name for metrics monitoring. (DEFAULT: )
NETHERMIND_MONITORING_JOB - Sets the job name for metrics monitoring. (DEFAULT: )
NETHERMIND_URL - Defines default URL for JSON RPC. (DEFAULT: )
```

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by dceleda** [comment]
I've added descriptions to the configs but we need confirm if all of them are possible to set via command line parameters.
Maybe some of them one can set only through environment variables.

File: src/Nethermind/Nethermind.Config/NoCategoryConfig.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public class NoCategoryConfig : INoCategoryConfig
         public string DataDir { get; set; }
         public string ConfigsDirectory { get; set; }
         public string BaseDbPath { get; set; }
-        public string LogLevel { get; set; }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Config/NoCategoryConfig.cs:15]: Why change that?

Comment (dceleda) [reply src/Nethermind/Nethermind.Config/NoCategoryConfig.cs:15]: Because it was wrong here. The command line option is actually "Log".

**Review by LukaszRozmej — APPROVED** [review]

**Review by kjazgar — APPROVED** [review]

**Comment by dceleda** [comment]
Implementing more changes...

File: src/Nethermind/Nethermind.Runner/Program.cs [hunk 2]
```diff
@@ -129,7 +130,7 @@ private static void Run(string[] args)
             IEnumerable<Type> configTypes = new TypeDiscovery().FindNethermindTypes(configurationType)
                 .Where(ct => ct.IsInterface);
 
-            foreach (Type configType in configTypes.OrderBy(c => c.Name))
+            foreach (Type configType in configTypes.Where(ct => !ct.IsAssignableTo(typeof(INoCategoryConfig))).OrderBy(c => c.Name))
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/Program.cs:133]: Can we move config related code to own methods?

File: src/Nethermind/Nethermind.Runner/Program.cs [hunk 3]
```diff
@@ -156,6 +157,25 @@ private static void Run(string[] args)
                 }
             }
 
+            Type noCategoryConfig = configTypes.FirstOrDefault(ct => ct.IsAssignableTo(typeof(INoCategoryConfig)));
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/Program.cs:160]: This too

**Pull request merged**